builder/frameworks/arduino/maple/stm32f4.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2014-present PlatformIO <contact@platformio.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Arduino
+
+Arduino Wiring-based Framework allows writing cross-platform software to
+control devices attached to a wide range of Arduino boards to create all
+kinds of creative coding, interactive objects, spaces or physical experiences.
+
+http://www.stm32duino.com
+"""
+
+from os.path import isdir, join
+
+Import("env")
+platform = env.PioPlatform()
+board = env.BoardConfig()
+
+FRAMEWORK_DIR = join(platform.get_package_dir(
+    "framework-arduinoststm32"), "STM32F4")
+assert isdir(FRAMEWORK_DIR)
+
+# default configuration values
+error_led_port = "GPIOA"
+error_led_pin = 7
+
+# remap board configuration values
+mcu_type = board.get("build.mcu")[:-2]
+if "stm32f407ve" in mcu_type:
+    ldscript = "jtag.ld"
+    variant = "generic_f407v"
+
+
+# upload related configuration remap
+# for all generic boards
+upload_protocol = env.subst("$UPLOAD_PROTOCOL")
+
+env.Append(
+    CFLAGS=[
+        "-mthumb",
+        "-MMD",
+        "-std=gnu11",
+        "-nostdlib",
+        "--param",
+        "max-inline-insns-single=500"],
+
+    CXXFLAGS=[
+        "-mthumb",
+        "-MMD",
+        "-std=gnu++11",
+        "-nostdlib",
+        "-fno-rtti",
+        "-fno-exceptions",
+        "--param",
+        "max-inline-insns-single=500"],
```

Comment:
For the same flags in C and C++, use `CCFLAGS`, here's an [example](https://github.com/platformio/platform-ststm32/blob/develop/builder/frameworks/arduino/maple/stm32f1.py#L101).
And flags like `mthumb`; `nostdlib`; `fno-rtti`; `fno-exceptions` are already defined in [build script](https://github.com/platformio/platform-ststm32/blob/develop/builder/main.py#L39). No need to include them in here.